### PR TITLE
store/copr: partly revert pr/35975, do it correctly this time

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #54969

Problem Summary:

### What changed and how does it work?

https://github.com/pingcap/tidb/pull/55155 try to revert  pr/35975 but not clean, coupling with https://github.com/pingcap/tidb/pull/41500 and bring bugs. PartitionNum is 1 even on non-partitioned table.

Do it correctly in this PR.
The new behaviour now:

- The 'select ... where ... order by ...' query will not reduce distsql concurrency to 2  (change)
- The simple 'select ... limit N' query, when N < 10000 , distsql concurrency is set to 1 (keep the behavior no change)
    - The same 'select ... limit N' query, on partitioned table, distsql concurrency is set to partition number (keep the behavior no change) 

### Check List

Tests 

- [ ] Unit test
- [X] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: tests/integrationtest/r/executor/issues.result [hunk 1]
```diff
@@ -868,3 +868,137 @@ set @@tidb_max_chunk_size = default;
...
+    └─Selection_11	100.00	0	cop[tikv]	NULL	NULL	eq(executor__issues.pt.val, 125)	<memory>	<disk>
+      └─TableFullScan_10	125.00	0	cop[tikv]	table:pt	NULL	keep order:false, stats:partial[val:missing]	<memory>	<disk>
+explain analyze select * from pt where val = 125 limit 100000;  # expected distsql concurrency 15
+id	estRows	actRows	task	access object	execution info	operator info	memory	disk
+Limit_8	204.80	0	root	NULL	NULL	offset:0, count:100000	<memory>	<disk>
+└─TableReader_13	204.80	0	root	partition:all	max_distsql_concurrency: 15	NULL	<memory>	<disk>
+  └─Limit_12	204.80	0	cop[tikv]	NULL	NULL	offset:0, count:100000	<memory>	<disk>
+    └─Selection_11	204.80	0	cop[tikv]	NULL	NULL	eq(executor__issues.pt.val, 125)	<memory>	<disk>
+      └─TableFullScan_10	256.00	0	cop[tikv]	table:pt	NULL	keep order:false, stats:partial[val:missing]	<memory>	<disk>
+explain analyze select * from pt order by id limit 100;         # expected distsql concurrency 7, but currently get 1, see issue #55190
```
Comment (tiancaiamao) [inline tests/integrationtest/r/executor/issues.result:986]: Known issue https://github.com/pingcap/tidb/issues/55190
Will not be fixed in this PR.

**Review by Defined2014 — APPROVED** [review]

**Review by windtalker — APPROVED** [review]
LGTM

**Comment by winoros** [comment]
Is 6.5 affected by this problem?

**Comment by tiancaiamao** [comment]
> Is 6.5 affected by this problem?

Yes

**Comment by tiancaiamao** [comment]
/test unit-test

https://github.com/pingcap/tidb/issues/55207

**Pull request merged**